### PR TITLE
MINOR: Fix TRACE logging in ReplicaManager

### PR DESCRIPTION
**Pull request by cmccabe**

### CODE REVIEW

**Comment by cmccabe** [comment]
The test failure is an Internal jenkins error.
```
17:56:31 Caused by: hudson.plugins.git.GitException: Error performing git command
17:56:31 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:2005)
17:56:31 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1964)
17:56:31 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1960)
17:56:31 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommand(CliGitAPIImpl.java:1597)
17:56:31 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl$5.execute(CliGitAPIImpl.java:770)
17:56:31 	... 11 more
17:56:31 Caused by: java.lang.OutOfMemoryError: unable to create new native thread
17:56:31 	at java.lang.Thread.start0(Native Method)
17:56:31 	at java.lang.Thread.start(Thread.java:717)
17:56:31 	at hudson.Proc$LocalProc.<init>(Proc.java:269)
17:56:31 	at hudson.Proc$LocalProc.<init>(Proc.java:218)
17:56:31 	at hudson.Launcher$LocalLauncher.launch(Launcher.java:930)
17:56:31 	at hudson.Launcher$ProcStarter.start(Launcher.java:450)
17:56:31 	at org.jenkinsci.plugins.gitclient.CliGitAPIImpl.launchCommandIn(CliGitAPIImpl.java:1992)
17:56:31 	... 15 more
17:56:31 ERROR: Error cloning remote repo 'origin'
```

File: core/src/main/scala/kafka/server/ReplicaManager.scala [hunk 1]
```diff
@@ -756,7 +756,8 @@ class ReplicaManager(val config: KafkaConfig,
           brokerTopicStats.allTopicsStats.messagesInRate.mark(numAppendedMessages)
 
           trace("%d bytes written to log %s-%d beginning at offset %d and ending at offset %d"
-            .format(records.sizeInBytes, topicPartition.topic, topicPartition.partition, info.firstOffset, info.lastOffset))
+            .format(records.sizeInBytes, topicPartition.topic, topicPartition.partition,
+              info.firstOffset.getOrElse(-1), info.lastOffset))
```
Comment (ijuma) [inline core/src/main/scala/kafka/server/ReplicaManager.scala]: Please change this to use string interpolation since it's safer.

**Comment by cmccabe** [comment]
Test failure is `kafka.server.DynamicBrokerReconfigurationTest.testAddRemoveSslListener`, not related

File: core/src/main/scala/kafka/server/ReplicaManager.scala [hunk 2]
```diff
@@ -754,8 +754,8 @@ class ReplicaManager(val config: KafkaConfig,
           brokerTopicStats.topicStats(topicPartition.topic).messagesInRate.mark(numAppendedMessages)
           brokerTopicStats.allTopicsStats.messagesInRate.mark(numAppendedMessages)
 
-          trace("%d bytes written to log %s-%d beginning at offset %d and ending at offset %d"
-            .format(records.sizeInBytes, topicPartition.topic, topicPartition.partition, info.firstOffset, info.lastOffset))
+          trace(s"${records.sizeInBytes} written to log ${topicPartition.topic}-${topicPartition.partition} " +
```
Comment (ijuma) [inline core/src/main/scala/kafka/server/ReplicaManager.scala]: `TopicPartition.toString()` does the right thing so you don't need to do it manually.

**Comment by cmccabe** [comment]
retest this please

**Review by ijuma — APPROVED** [review]
LGTM

**Pull request merged**